inc/Menu/Item/Revert.php

```diff
@@ -7,20 +7,33 @@
  *
  * Quick revert to the currently shown page revision
  */
-class Revert extends AbstractItem {
+class Revert extends AbstractItem
+{
 
     /** @inheritdoc */
-    public function __construct() {
+    public function __construct()
+    {
         global $REV;
         global $INFO;
+        global $INPUT;
         parent::__construct();
 
-        if(!$INFO['ismanager'] || !$REV || !$INFO['writable']) {
+        // if we are comparing with the current version then
+        // assume the other version is what we want to revert to
+        $comparedWithCurrent = in_array('current', $INPUT->ref('rev2')) || in_array('', $INPUT->ref('rev2'));
+        if ($comparedWithCurrent) {
+            foreach ($INPUT->ref('rev2') as $revision) {
+                if ($revision !== 'current' && $revision !== '') {
+                    $REV = $revision;
```

Comment:
Well, there are two approaches here:

1. Fix global `$REV` to make sure if we are comparing with current revision, it has a value, which is what you are doing here. You may want to make sure the patch works nicely with other code that processes `$REV` e.g. https://github.com/splitbrain/dokuwiki/blob/7a0b5f30c2d63a8e403568877b6f3f07db70bdd6/inc/html.php#L1224-L1237
Menu/Item is not the proper place for this to happen. I think one of the better places would be here https://github.com/splitbrain/dokuwiki/blob/7a0b5f30c2d63a8e403568877b6f3f07db70bdd6/inc/common.php#L231-L245
2. Or, once you find out `$comparedWithCurrent == true`, don't change `$REV` but just make sure \RuntimeException is not being raised.

Reply:
It seems like 2 would be the simplest. Just remove everything but `$comparedWithCurrent = in_array('current', $INPUT->ref('rev2')) || in_array('', $INPUT->ref('rev2'));` and then changed the check to:

```php
if (!$INFO['ismanager'] || !$REV || !$INFO['writable'] || !$comparedWithCurrent) {
    throw new \RuntimeException('revert not available');
}
```

Reply:
Exactly. Do you mind committing this change to your branch?

Reply:
Just committed those changes.